pkgs/development/libraries/nghttp2/default.nix

```diff
@@ -1,65 +1,72 @@
-{ lib, stdenv, fetchurl, pkg-config
+{ lib
+, stdenv
+, fetchurl
+, installShellFiles
+, pkg-config
 
-# Optional Dependencies
-, openssl ? null, zlib ? null
-, enableLibEv ? !stdenv.hostPlatform.isWindows, libev ? null
-, enableCAres ? !stdenv.hostPlatform.isWindows, c-ares ? null
-, enableHpack ? false, jansson ? null
+# Optional dependencies
+, enableApp ? with stdenv.hostPlatform; !isWindows && !isStatic
+, c-ares ? null, libev ? null, openssl ? null, zlib ? null
```

Comment:
```suggestion
, c-ares, libev, openssl, zlib
```

Reply:
Is there an actual issue here? It's clearer as it is.
We want some inputs to be void and ignored in some cases. Such as building only the lib to bootstrap fetchurl.

Reply:
yes, it hides bugs. bootstrapping is a bummer so we need to keep some of them.

Reply:
Wether you like it or not but `? null` hides bugs especially around aliases and removed attrs.

Reply:
You use the function always with `callPackage`, so the null defaults never get used there.

Reply:
> You use the function always with `callPackage`, 

What function?

> so the null defaults never get used there.

So it would just be equivalent one way or the other?

> EDIT: and if your point is just docs, the comment that's there seems a better approach.

Sorry I don't understand this one. What comment?

---

This discussion has been awfully vague for me up to now. I still have no idea what _actual issue_ we are talking about.
And if there an issue with `? null` in `nghttp2` derivation, why has no one opened a ticket for it or come up with a PR to "solve" it until before? Why is this a thing _now_?

My main concern is: I don't want this to stall over something that's bigger than this particular derivation and somewhat out of scope for this PR which is mainly about the version update. There is enough such abandoned PRs lying around the repo.

Reply:
The whole file defines a function.  That's the function I meant.

Anyway, I agree these `null` defaults can be changed separately; maybe it's even better that way.  (and I don't consider it important)

Reply:
Thank you for the clarification. :)